apps/prairielearn/src/models/course-instances.ts

```diff
@@ -0,0 +1,52 @@
+import { callValidatedOneRow } from '@prairielearn/postgres';
+import { z } from 'zod';
+import { IdSchema } from '../lib/db-types';
+import { idsEqual } from '../lib/id';
+
+const CourseInstanceAuthzSchema = z.object({
+  course_instances: z.array(
+    z.object({
+      short_name: z.string().nullable(),
+      long_name: z.string().nullable(),
+      id: IdSchema,
+      formatted_start_date: z.string(),
+      formatted_end_date: z.string(),
+      has_course_instance_permission_view: z.boolean(),
+    }),
+  ),
+});
+
+export async function selectAuthorizedCourseInstancesForCourse({
+  course_id,
+  user_id,
+  authn_user_id,
+  is_administrator,
+}: {
+  course_id: string;
+  user_id: string;
+  authn_user_id: string;
+  is_administrator: boolean;
+}) {
+  const { course_instances: authnCourseInstances } = await callValidatedOneRow(
+    'course_instances_with_staff_access',
```

Comment:
This sproc (and `courses_with_staff_access`) are basically ready to be turned into standard queries. I'm opting to leave that for a future PR so that it can be reviewed in isolation.